File: java-package/brainflow/src/main/java/brainflow/BrainFlowModelParams.java
package brainflow;

import com.google.gson.Gson;

/**
 * describe model parameters
 */
public class BrainFlowModelParams
{
    public int metric;
    public int classifier;
    public String file;
    public String other_info;

    public BrainFlowModelParams (int metric, int classifier)
    {
        this.metric = metric;
        this.classifier = classifier;
        this.file = "";
        this.other_info = "";
    }

    public int get_metric ()
    {
        return metric;
    }

    public void set_metric (int metric)
    {
        this.metric = metric;
    }

    public int get_classifier ()
    {
        return classifier;
    }

    public void set_classifier (int classifier)
    {
        this.classifier = classifier;
    }

    public String get_file ()
    {
        return file;
    }

    public void set_file (String file)
    {
        this.file = file;
    }

    public String get_other_info ()
    {
        return other_info;
    }

    public void set_other_info (String other_info)
    {
        this.other_info = other_info;
    }

    public String to_json ()
    {
        return new Gson ().toJson (this);
    }

}

File: csharp-package/brainflow/brainflow/brainflow_model_params.cs
using System.Collections.Generic;
using System.IO;
using System.Runtime.Serialization;
using System.Runtime.Serialization.Json;
using System.Text;


namespace brainflow
{
    /// <summary>
    ///  Describe model
    /// </summary>
    [DataContract]
    public class BrainFlowModelParams
    {
        /// <summary>
        /// metric to caclulate
        /// </summary>
        [DataMember]
        public int metric;
        /// <summary>
        /// classifier to use
        /// </summary>
        [DataMember]
        public int classifier;
        /// <summary>
        /// path to model file
        /// </summary>
        [DataMember]
        public string file;
        /// <summary>
        /// other info
        /// </summary>
        [DataMember]
        public string other_info;

        public BrainFlowModelParams (int metric, int classifier)
        {
            this.metric = metric;
            this.classifier = classifier;
            file = "";
            other_info = "";
        }

        public string to_json ()
        {
            // copypaste from https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/how-to-serialize-and-deserialize-json-data?redirectedfrom=MSDN
            var ms = new MemoryStream ();
            var ser = new DataContractJsonSerializer (typeof (BrainFlowModelParams));
            ser.WriteObject (ms, this);
            byte[] json = ms.ToArray ();
            ms.Close ();
            return Encoding.UTF8.GetString (json, 0, json.Length);
        }
    }
}
